### PR TITLE
[Feature] Cloud compare-reports command support selector

### PR DESCRIPTION
**Pull request by kentwelcome**
Signed-off-by: Kent Huang <kent@infuseai.io>

**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Enhancement

**What this PR does / why we need it**:
 - If no base or target provides when calling `cloud compare-reports` command, will use the report selector to select uploaded reports from the cloud
 CleanShot 2023-02-02 at 16 50 31

**Which issue(s) this PR fixes**:
n/a

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

**Comment by kentwelcome** [comment]
The legacy project list API response:
CleanShot 2023-02-03 at 15 54 56

File: piperider_cli/cloud_connector.py [hunk 1]
```diff
@@ -281,8 +318,19 @@ def compare_reports(base=None, target=None, tables_from='all', summary_file=None
             console.rule('Please login PipeRider Cloud first', style='red')
             return 1
 
-        base_id = get_run_report_id(base)
-        target_id = get_run_report_id(target)
+        if base is None or target is None:
+            base, target = select_cloud_reports(base=base, target=target)
```
Comment (wcchang1115) [inline piperider_cli/cloud_connector.py:322]: Just a discussion:
what do you think if `select_cloud_reports` returns cloud report id directly?

File: piperider_cli/compare_report.py [hunk 2]
```diff
@@ -648,7 +651,7 @@ def _report_validater(answers, current) -> bool:
             else:
                 return len(current) == limit
 
-        profiler_outputs = self.list_existing_outputs()
+        profiler_outputs = self.profiler_outputs if self.profiler_outputs else self.list_existing_outputs()
```
Comment (wcchang1115) [inline piperider_cli/compare_report.py:654]: found a problem here when there are no cloud reports on the cloud, it tries to list existing local reports
1. it should not list local reports when for user when exec `piperider cloud compare-reports`
2. exception happened when `profiler_output_path` is not specified in listing existing outputs

**Review by wcchang1115 — CHANGES_REQUESTED** [review]
Hi Kent,
I think we need to check reports selection behavior, thank you!

File: piperider_cli/cloud_connector.py (hunk 1 above)
Comment (kentwelcome) [reply piperider_cli/cloud_connector.py:322]: In this function, I think return report id should be more reasonable. I will also rename the function `select_cloud_reports` to `select_cloud_report_ids`

Comment (kentwelcome) [reply piperider_cli/cloud_connector.py]: Fix

**Review by kentwelcome — COMMENTED** [review]
- select cloud report function return report id directly
 - Fix `self.profiler_outputs` exception issue when cloud project doesn't
   have any uploaded report

File: piperider_cli/compare_report.py (hunk 2 above)
Comment (kentwelcome) [reply piperider_cli/compare_report.py]: Fixed

**Review by wcchang1115 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**